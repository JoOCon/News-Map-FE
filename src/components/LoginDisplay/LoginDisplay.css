.placement-container {
  height: 110%;
  width: 100%;
  display: flex;
  position: absolute;
  margin: -4rem 0 0rem 0;
  justify-content: center;
  align-items: center;
  background: url("../../images/worldpoly.gif");
  background-repeat: no-repeat;
  background-size: 90%;
  background-position: center;
  background-color: #0c0c0c; }

.login-container {
  display: flex;
  flex-direction: column;
  align-items: center;
  background: rgba(170, 170, 170, 0.9);
  padding: 0.3em 1em;
  border-radius: 0.3em;
  box-shadow: 0.1em 0.2em 0.5em #646464; }

.login-container h1 {
  margin: 0;
  font-size: 3em;
  color: white; }

.login-button {
  font-family: 'Roboto Slab', serif;
  margin: 0.5em;
  padding: 0.2em 0.5em;
  background-color: #646464;
  border-radius: 0.3em;
  border: none;
  color: white;
  font-size: 2em;
  cursor: pointer; }

.login-button:hover {
  animation: login-animation 1.5s linear infinite; }

@media only screen and (max-width: 600px) {
  .placement-container {
    background-size: 140%; }
  .login-container {
    padding: 0 .3em;
    border-radius: 0.3em; }
  .login-button {
    margin: 0 0 .5em 0; } }

@keyframes login-animation {
  0% {
    box-shadow: 0 0 0 0 rgba(100, 100, 100, 0.5); }
  50% {
    box-shadow: 0 0 0 4px rgba(0, 220, 0, 0.5); }
  100% {
    box-shadow: 0 0 0 0 rgba(100, 100, 100, 0.5); } }
