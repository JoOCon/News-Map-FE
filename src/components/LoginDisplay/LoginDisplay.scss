.placement-container {
  height: 110%;
  width: 100%;
  display: flex;
  position: absolute;
  margin: -4rem 0 0rem 0;
  justify-content: center;
  align-items: center;
  background: url('../../images/worldpoly.gif');
  background-repeat: no-repeat;
  background-size: 70%;
  background-position: center;
  background-color: #0c0c0c;
}

.login-container {
  display: flex;
  flex-direction: column;
  align-items: center;
  background: rgba(191, 186, 183, 0.9);
  padding: 0.3em 1em;
  border-radius: 0.3em;
  box-shadow: 0.1em 0.2em rgba(100, 100, 100, 1);
}

.login-container h1 {
  margin: 0;
  font-size: 3em;
  color: rgb(255, 255, 255);
}

.login-button {
  font-family: 'Roboto Slab', serif;
  margin: 0.5em;
  padding: 0.2em 0.5em;
  background-color: #646464;
  border: 1px solid transparent;
  border-radius: 0.3em;
  color: rgb(255, 255, 255);
  font-size: 2em;
  cursor: pointer;
}

.login-button:hover {
  border: 1px solid rgba(255, 215, 50, 1);
  transition: 500ms;
}
  