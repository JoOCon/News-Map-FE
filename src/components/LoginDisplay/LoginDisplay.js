import React from 'react';
import PropTypes from 'prop-types';
import { GoogleLogin } from 'react-google-login';
import './LoginDisplay.css';

export const LoginDisplay = ({ login }) => (
  <div className="placement-container">
    <div className="login-container">
      <h1>EVENT MAPPER</h1>
      <GoogleLogin
        clientId={process.env.REACT_APP_GOOGLE_CLIENT_ID}
        className="login-button"
        buttonText="Login with Google"
        onSuccess={login}
        onFailure={login}
      />
    </div>
  </div>
);

LoginDisplay.propTypes = {
  login: PropTypes.func
};
