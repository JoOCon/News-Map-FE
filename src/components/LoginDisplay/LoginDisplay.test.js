import React from 'react';
import { LoginDisplay } from './LoginDisplay';

describe('LoginDisplay', () => {
  let wrapper = shallow(<LoginDisplay login={jest.fn()} />);
  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
