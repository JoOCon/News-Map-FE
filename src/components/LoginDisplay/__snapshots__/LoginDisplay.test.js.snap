// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginDisplay should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LoginDisplay
    login={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="login-container"
      >
        <h1>
          EVENT MAPPER
        </h1>
        <t
          accessType="online"
          buttonText="Login with Google"
          className="login-button"
          cookiePolicy="single_host_origin"
          disabledStyle={
            Object {
              "opacity": 0.6,
            }
          }
          fetchBasicProfile={true}
          isSignedIn={false}
          jsSrc="https://apis.google.com/js/client:platform.js"
          onFailure={[MockFunction]}
          onRequest={[Function]}
          onSuccess={[MockFunction]}
          prompt=""
          scope="profile email"
          tag="button"
          type="button"
          uxMode="popup"
        />
      </div>,
      "className": "placement-container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            EVENT MAPPER
          </h1>,
          <t
            accessType="online"
            buttonText="Login with Google"
            className="login-button"
            cookiePolicy="single_host_origin"
            disabledStyle={
              Object {
                "opacity": 0.6,
              }
            }
            fetchBasicProfile={true}
            isSignedIn={false}
            jsSrc="https://apis.google.com/js/client:platform.js"
            onFailure={[MockFunction]}
            onRequest={[Function]}
            onSuccess={[MockFunction]}
            prompt=""
            scope="profile email"
            tag="button"
            type="button"
            uxMode="popup"
          />,
        ],
        "className": "login-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "EVENT MAPPER",
          },
          "ref": null,
          "rendered": "EVENT MAPPER",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessType": "online",
            "buttonText": "Login with Google",
            "className": "login-button",
            "clientId": undefined,
            "cookiePolicy": "single_host_origin",
            "disabledStyle": Object {
              "opacity": 0.6,
            },
            "fetchBasicProfile": true,
            "isSignedIn": false,
            "jsSrc": "https://apis.google.com/js/client:platform.js",
            "onFailure": [MockFunction],
            "onRequest": [Function],
            "onSuccess": [MockFunction],
            "prompt": "",
            "scope": "profile email",
            "tag": "button",
            "type": "button",
            "uxMode": "popup",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="login-container"
        >
          <h1>
            EVENT MAPPER
          </h1>
          <t
            accessType="online"
            buttonText="Login with Google"
            className="login-button"
            cookiePolicy="single_host_origin"
            disabledStyle={
              Object {
                "opacity": 0.6,
              }
            }
            fetchBasicProfile={true}
            isSignedIn={false}
            jsSrc="https://apis.google.com/js/client:platform.js"
            onFailure={[MockFunction]}
            onRequest={[Function]}
            onSuccess={[MockFunction]}
            prompt=""
            scope="profile email"
            tag="button"
            type="button"
            uxMode="popup"
          />
        </div>,
        "className": "placement-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1>
              EVENT MAPPER
            </h1>,
            <t
              accessType="online"
              buttonText="Login with Google"
              className="login-button"
              cookiePolicy="single_host_origin"
              disabledStyle={
                Object {
                  "opacity": 0.6,
                }
              }
              fetchBasicProfile={true}
              isSignedIn={false}
              jsSrc="https://apis.google.com/js/client:platform.js"
              onFailure={[MockFunction]}
              onRequest={[Function]}
              onSuccess={[MockFunction]}
              prompt=""
              scope="profile email"
              tag="button"
              type="button"
              uxMode="popup"
            />,
          ],
          "className": "login-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "EVENT MAPPER",
            },
            "ref": null,
            "rendered": "EVENT MAPPER",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessType": "online",
              "buttonText": "Login with Google",
              "className": "login-button",
              "clientId": undefined,
              "cookiePolicy": "single_host_origin",
              "disabledStyle": Object {
                "opacity": 0.6,
              },
              "fetchBasicProfile": true,
              "isSignedIn": false,
              "jsSrc": "https://apis.google.com/js/client:platform.js",
              "onFailure": [MockFunction],
              "onRequest": [Function],
              "onSuccess": [MockFunction],
              "prompt": "",
              "scope": "profile email",
              "tag": "button",
              "type": "button",
              "uxMode": "popup",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
